File: app/src/main/java/com/cappielloantonio/play/subsonic/models/Playlist.java
package com.cappielloantonio.play.subsonic.models;

import com.tickaroo.tikxml.annotation.Attribute;
import com.tickaroo.tikxml.annotation.Element;
import com.tickaroo.tikxml.annotation.Xml;
import com.tickaroo.tikxml.converters.date.rfc3339.DateRfc3339TypeConverter;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Xml
public class Playlist {
    protected List<String> allowedUsers;
    @Attribute
    protected String id;
    @Attribute
    protected String name;
    @Attribute
    protected String comment;
    @Attribute
    protected String owner;
    @Attribute(name = "public")
    protected Boolean universal;
    @Attribute
    protected int songCount;
    @Attribute
    protected int duration;
    @Attribute(converter = DateRfc3339TypeConverter.class)
    protected Date created;
    @Attribute(converter = DateRfc3339TypeConverter.class)
    protected Date changed;
    @Attribute
    protected String coverArtId;
    
    public List<String> getAllowedUsers() {
        if (allowedUsers == null) {
            allowedUsers = new ArrayList<>();
        }
        return this.allowedUsers;
    }

    public void setAllowedUsers(List<String> allowedUsers) {
        this.allowedUsers = allowedUsers;
    }

    public String getId() {
        return id;
    }
    
    public void setId(String value) {
        this.id = value;
    }

    public String getName() {
        return name;
    }
    
    public void setName(String value) {
        this.name = value;
    }

    public String getComment() {
        return comment;
    }

    public void setComment(String value) {
        this.comment = value;
    }

    public String getOwner() {
        return owner;
    }
    
    public void setOwner(String value) {
        this.owner = value;
    }

    public Boolean isUniversal() {
        return universal;
    }

    public void setUniversal(Boolean value) {
        this.universal = value;
    }

    public int getSongCount() {
        return songCount;
    }
    
    public void setSongCount(int value) {
        this.songCount = value;
    }
    
    public int getDuration() {
        return duration;
    }
    
    public void setDuration(int value) {
        this.duration = value;
    }
    
    public Date getCreated() {
        return created;
    }

    public void setCreated(Date value) {
        this.created = value;
    }
    
    public Date getChanged() {
        return changed;
    }

    public void setChanged(Date value) {
        this.changed = value;
    }

    public String getCoverArtId() {
        return coverArtId;
    }

    public void setCoverArtId(String value) {
        this.coverArtId = value;
    }
}

File: app/src/main/java/com/cappielloantonio/play/subsonic/models/Playlist.kt
package com.cappielloantonio.play.subsonic.models

import android.os.Parcelable
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize
import java.util.*

@Parcelize
@Entity(tableName = "playlist")
open class Playlist(
    @PrimaryKey
    @ColumnInfo(name = "id")
    open var id: String
) : Parcelable {
    @ColumnInfo(name = "name")
    var name: String? = null

    @Ignore
    var comment: String? = null

    @Ignore
    var owner: String? = null

    @Ignore
    @SerializedName("public")
    var isUniversal: Boolean? = null

    @Ignore
    var songCount: Int = 0

    @ColumnInfo(name = "duration")
    var duration: Long = 0

    @Ignore
    var created: Date? = null

    @Ignore
    var changed: Date? = null

    @ColumnInfo(name = "coverArt")
    var coverArtId: String? = null

    @Ignore
    var allowedUsers: List<String>? = null
}